packages/lib/apps/getEnabledApps.ts

```diff
@@ -2,12 +2,47 @@ import getApps from "@calcom/app-store/utils";
 import type { CredentialData } from "@calcom/app-store/utils";
 import { prisma } from "@calcom/prisma";
 
-const getEnabledApps = async (userCredentials: CredentialData[]) => {
+/**
+ *
+ * @param credentials - Can be user or team credentials
+ * @param filterOnCredentials - Only include apps where credentials are present
+ * @returns A list of enabled app metadata & credentials tied to them
+ */
+const getEnabledApps = async (credentials: CredentialData[], filterOnCredentials?: boolean) => {
+  const filterOnIds = {
+    credentials: {
+      some: {
+        OR: [],
+      },
+    },
+  };
```

Comment:
The filter query to only include apps that have credentials that contain the `userId` or `teamId` present in credentials passed.